### PR TITLE
pipeline-lite: Introduce metadata component to pipeline-lite

### PR DESCRIPTION
**Pull request by dushyanthsc**
This change introduces the metadata component to pipeline-liteinstallation. With the change pipeline-lite installation:

  1. Does not include metadata-ui
  2. mysql installation is used instead of metadata-db
  3. Replica count has been reduced 1 instead of 3

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/pipelines/1840)

### CODE REVIEW

**Comment by gaoning777** [comment]
Have you validated that the lightweight deployment successfully deploy the metadata service?

**Comment by dushyanthsc** [comment]
> Have you validated that the lightweight deployment successfully deploy the metadata service?

Yes, I validated the deployment. Used the curl commands in here https://www.kubeflow.org/docs/components/misc/metadata/#backend-and-rest-api on the metadata backend pod to verify that I can create a type and list it.

File: manifests/kustomize/base/pipeline/kustomization.yaml [hunk 1]
```diff
@@ -26,4 +26,6 @@ resources:
 - ml-pipeline-viewer-crd-sa.yaml
 - pipeline-runner-role.yaml
 - pipeline-runner-rolebinding.yaml
-- pipeline-runner-sa.yaml
\ No newline at end of file
+- pipeline-runner-sa.yaml
```
Comment (IronPan) [inline manifests/kustomize/base/pipeline/kustomization.yaml]: please add another folder under base/ named metadata for metadata components.

File: manifests/kustomize/base/pipeline/metadata-deployment.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    metadata:
+      labels:
+        component: metadata-server
+    spec:
+      containers:
+      - name: container
+        image: gcr.io/kubeflow-images-public/metadata:v0.1.8
+        command: ["./server/server",
+                  "--http_port=8080",
+                  "--mysql_service_host=mysql",
```
Comment (IronPan) [inline manifests/kustomize/base/pipeline/metadata-deployment.yaml]: mysql host/port shouldn't be passed in as parameter.
It should be inferred from metadata pod's env variable 
Example
https://github.com/kubeflow/pipelines/blob/master/backend/src/apiserver/client_manager.go#L46

File: manifests/kustomize/base/pipeline/kustomization.yaml (hunk 1 above)
Comment (dushyanthsc) [reply manifests/kustomize/base/pipeline/kustomization.yaml]: Done.

File: manifests/kustomize/base/pipeline/metadata-deployment.yaml (hunk 2 above)
Comment (dushyanthsc) [reply manifests/kustomize/base/pipeline/metadata-deployment.yaml]: As per discussion, Keeping this as is for the moment.

**Pull request merged**